pkg/controller/installation/products/mobiledeveloperconsole/reconciler.go

```diff
@@ -0,0 +1,162 @@
+
+	return &Reconciler{
+		ConfigManager: configManager,
+		Config:        config,
+		mpm:           mpm,
+		logger:        logger,
+		Reconciler:    resources.NewReconciler(mpm),
+	}, nil
+}
+
+func (r *Reconciler) Reconcile(ctx context.Context, inst *v1alpha1.Installation, product *v1alpha1.InstallationProductStatus, client pkgclient.Client) (v1alpha1.StatusPhase, error) {
+
+	phase, err := r.ReconcileNamespace(ctx, r.Config.GetNamespace(), inst, client)
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+
+	version, err := resources.NewVersion(v1alpha1.OperatorVersionMDC)
+	if err != nil {
+		return v1alpha1.PhaseFailed, errors.Wrap(err, "invalid version number for mdc")
+	}
+
+	phase, err = r.ReconcileSubscription(
+		ctx,
+		inst,
+		marketplace.Target{Pkg: defaultSubscriptionName, Channel: marketplace.IntegreatlyChannel, Namespace: r.Config.GetNamespace()},
+		client,
+		version,
+	)
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+
+	phase, err = r.reconcileCustomResource(ctx, client)
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+
+	phase, err = r.handleProgress(ctx, client)
+	if err != nil || phase != v1alpha1.PhaseCompleted {
+		return phase, err
+	}
+
+	product.Host = r.Config.GetHost()
+	product.Version = r.Config.GetProductVersion()
+
+	r.logger.Infof("%s has reconciled successfully", r.Config.GetProductName())
+	return v1alpha1.PhaseCompleted, nil
+}
+
+func (r *Reconciler) GetPreflightObject(ns string) runtime.Object {
+	return nil
+}
+
+func (r *Reconciler) reconcileCustomResource(ctx context.Context, client pkgclient.Client) (v1alpha1.StatusPhase, error) {
+	r.logger.Info("reconciling mobile-developer-console custom resource")
+
+	cr := &mdc.MobileDeveloperConsole{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: r.Config.GetNamespace(),
+			Name:      resourceName,
+		},
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: mdc.SchemeGroupVersion.String(),
+			Kind:       mdc.MobileDeveloperConsoleKind,
+		},
+		Spec: mdc.MobileDeveloperConsoleSpec{
+			OAuthClientID:     "mobile-developer-console",
+			OAuthClientSecret: "foobar",
+		},
+	}
+
+	// attempt to create the custom resource
+	if err := client.Create(ctx, cr); err != nil && !k8serr.IsAlreadyExists(err) {
+		return v1alpha1.PhaseFailed, errors.Wrap(err, "failed to get or create a mdc custom resource")
+	}
+
+	// if there are no errors, the phase is complete
+	return v1alpha1.PhaseCompleted, nil
+}
+
+func (r *Reconciler) handleProgress(ctx context.Context, client pkgclient.Client) (v1alpha1.StatusPhase, error) {
+	r.logger.Info("checking mdc pods are running")
+
+	pods := &corev1.PodList{}
+	err := client.List(ctx, &pkgclient.ListOptions{Namespace: r.Config.GetNamespace()}, pods)
+	if err != nil {
+		return v1alpha1.PhaseFailed, errors.Wrap(err, "failed to list mdc in mdc namespace")
+	}
+
+	// expecting 2 pods in total
+	if len(pods.Items) < 2 {
```

Comment:
if this is how it has to be, then so be it, but it'd be preferable to check it was ready from the status object of the CR we created. If that is not possible, we should feed that back to the operator development team so that they can add it.

Reply:
I have removed this